### PR TITLE
Rename trigger name userError and update documentation.

### PR DESCRIPTION
**Pull request by tiendt**
For #11084

### CODE REVIEW

**Comment by tiendt** [comment]
@ericlindley-g can you take a look at the description of the new error trigger? Thanks!

**Comment by ericlindley-g** [comment]
Looks good to me (and I care about the general feature as it relates to content quality), but I think @rudygalfi would be a better reviewer on the PM side as the PM lead for AMP analytics

/cc @bpaduch as well

File: extensions/amp-analytics/amp-analytics.md [hunk 1]
```diff
@@ -352,6 +352,21 @@ Notice that selector can be used to only specify a single element, not a collect
 The element visibility trigger waits for the signal specified by the `waitFor` property in `visibilitySpec` before tracking element visibility. If `waitFor` is not specified, it waits for element's [`ini-load`](#initial-load-trigger) signal. See `waitFor` docs for more details.
 
 
+##### Error trigger
+
+The user error event (`"on": "user-error"`) is triggered when there are errors that belong to publishers. This includes, but not limited to, mis-config of an AMP component, mis-configured ads, or failed assertions,
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: mis-config -> misconfiguration
mis-configured -> misconfigured

File: examples/analytics-error-reporting.amp.html [hunk 2]
```diff
@@ -72,12 +72,12 @@ <h2 id="amp-iframe">IFrame</h2>
     <script type="application/json">
         {
             "requests": {
-                "user-error": "http://localhost:8000/dsdsdsd"
+                "error": "http://localhost:8000/dsdsdsd"
```
Comment (rudygalfi) [inline examples/analytics-error-reporting.amp.html]: Is it possible to have this request include detail on what the error is? If so, then it should be included as a substitution variable and documented.

File: extensions/amp-analytics/amp-analytics.md [hunk 3]
```diff
@@ -219,7 +219,7 @@ The `extraUrlParamsReplaceMap` attribute specifies a map of keys and values that
 
 The `triggers` configuration object describes when an analytics request should be sent. The `triggers` attribute contains a key-value pair of trigger-name and  trigger-configuration. A trigger-name can be any string comprised of alphanumeric characters (a-zA-Z0-9). Triggers from a  configuration with lower precedence are overridden by triggers with the same names from a configuration with higher precedence.
 
-  - `on` (required) The event to listener for. Valid values are `render-start`, `ini-load`, `visible`, `click`, `scroll`, `timer`, and `hidden`.
+  - `on` (required) The event to listener for. Valid values are `render-start`, `ini-load`, `visible`, `click`, `scroll`, `timer`, `hidden`, and `user-error`.
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: Can you change to "the event to listen for"?

File: extensions/amp-analytics/amp-analytics.md (hunk 1 above)
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: is triggered when an error occurs that is attributable to the author of the page or to software that is used in publishing the page. ... User errors are also reported in the developer console.

File: examples/analytics-error-reporting.amp.html (hunk 2 above)
Comment (tiendt) [reply examples/analytics-error-reporting.amp.html]: I think so yeah, let me try that and let you know!

File: extensions/amp-analytics/amp-analytics.md [hunk 4]
```diff
@@ -219,7 +219,7 @@ The `extraUrlParamsReplaceMap` attribute specifies a map of keys and values that
 
 The `triggers` configuration object describes when an analytics request should be sent. The `triggers` attribute contains a key-value pair of trigger-name and  trigger-configuration. A trigger-name can be any string comprised of alphanumeric characters (a-zA-Z0-9). Triggers from a  configuration with lower precedence are overridden by triggers with the same names from a configuration with higher precedence.
 
-  - `on` (required) The event to listener for. Valid values are `render-start`, `ini-load`, `visible`, `click`, `scroll`, `timer`, and `hidden`.
+  - `on` (required) The event to listene for. Valid values are `render-start`, `ini-load`, `visible`, `click`, `scroll`, `timer`, `hidden`, and `user-error`.
```
Comment (zhouyx) [inline extensions/amp-analytics/amp-analytics.md]: nit: typo `listen`

File: extensions/amp-analytics/amp-analytics.md [hunk 5]
```diff
@@ -352,6 +352,23 @@ Notice that selector can be used to only specify a single element, not a collect
 The element visibility trigger waits for the signal specified by the `waitFor` property in `visibilitySpec` before tracking element visibility. If `waitFor` is not specified, it waits for element's [`ini-load`](#initial-load-trigger) signal. See `waitFor` docs for more details.
 
 
+##### Error trigger
+
+The user error event (`"on": "user-error"`) is triggered when an error occurs that is attributable to the author of the page or to software that is used in publishing the page. This includes, but not limited to, misconfiguration of an AMP component, misconfigured ads, or failed assertions. User errors are also reported in the developer console.
```
Comment (zhouyx) [inline extensions/amp-analytics/amp-analytics.md]: to @rudygalfi 
I know we collect user error from both 1p and 3p. Do we want to identify them with different event name later? One possible usage is that publishers can sample 1p error and 3p error at different sample rate?

Comment (rudygalfi) [reply extensions/amp-analytics/amp-analytics.md]: Perhaps not necessary at this stage, but yes, a variable for error type (1P vs 3P) and the possibility of a userErrorSpec that could enable different sampling strategies would be good future considerations.

File: extensions/amp-analytics/amp-analytics.md [hunk 6]
```diff
@@ -352,6 +352,23 @@ Notice that selector can be used to only specify a single element, not a collect
...
+
+Currently, the trigger is not stable with excluding errors generated from A4A iframe embed, therefore, unwanted errors maybe reported.
```
[the remark this answers is not in the document]
Comment (tiendt) [reply extensions/amp-analytics/amp-analytics.md]: @rudygalfi do you know if there's a doc/link talking about A4A iframe embed?

File: spec/amp-var-substitutions.md [hunk 7]
```diff
@@ -546,6 +548,22 @@ Provides the absolute Y coordinate of the top edge of the element specified by `
 * **platform variable**: N/A
 * **amp-analytics variable**: `${elementY}`
 
+#### Error Message
+
+Provides the message of the error that triggered an user error event. This variable is only available in a `trigger` of type `user-error`
+
+* **platform variable**: N/A
+* **amp-analytics variable**: `${errorMessage}`
+  * Example: `"error": "http://localhost:8000/message=${errorMessage}"`
```
Comment (rudygalfi) [inline spec/amp-var-substitutions.md]: Example would be something like an expected string value for an error.

File: spec/amp-var-substitutions.md [hunk 8]
```diff
@@ -546,6 +548,22 @@ Provides the absolute Y coordinate of the top edge of the element specified by `
...
+
+#### Error Name
+
+Provides the name of the error that triggered an user error event. This variable is only available in a `trigger` of type `user-error`
+
+* **platform variable**: N/A
+* **amp-analytics variable**: `${errorName}`
+  * Example : `"error": "http://localhost:8000/name=${errorName}"`
```
Comment (rudygalfi) [inline spec/amp-var-substitutions.md]: Example would be something like an expected string value for an error.

File: spec/amp-var-substitutions.md (hunk 7 above)
Comment (rudygalfi) [reply spec/amp-var-substitutions.md]: Also, nit: Should say "Example value" to be consistent with other entries on the page.

File: spec/amp-var-substitutions.md (hunk 8 above)
Comment (rudygalfi) [reply spec/amp-var-substitutions.md]: Also, nit: Should say "Example value" to be consistent with other entries on the page, and get rid of space before the colon.

File: examples/analytics-error-reporting.amp.html (hunk 2 above)
Comment (rudygalfi) [reply examples/analytics-error-reporting.amp.html]: Can the request be updated to something like

`http://localhost:8000/ping?en=${errorName}&em=${errorMessage}`

File: extensions/amp-analytics/amp-analytics.md [hunk 9]
```diff
@@ -219,7 +219,7 @@ The `extraUrlParamsReplaceMap` attribute specifies a map of keys and values that
 
 The `triggers` configuration object describes when an analytics request should be sent. The `triggers` attribute contains a key-value pair of trigger-name and  trigger-configuration. A trigger-name can be any string comprised of alphanumeric characters (a-zA-Z0-9). Triggers from a  configuration with lower precedence are overridden by triggers with the same names from a configuration with higher precedence.
 
-  - `on` (required) The event to listener for. Valid values are `render-start`, `ini-load`, `click`, `scroll`, `timer`, `visible`, `hidden`, [`access-*`](../amp-access/amp-access-analytics.md), and [`video-*`](./amp-video-analytics.md)
+  - `on` (required) The event to listener for. Valid values are `render-start`, `ini-load`, `click`, `scroll`, `timer`, `visible`, `hidden`, `user-error`, [`access-*`](../amp-access/amp-access-analytics.md), and [`video-*`](./amp-video-analytics.md)
```
Comment (rudygalfi) [inline extensions/amp-analytics/amp-analytics.md]: nit: Could you fix so it's "to listen for"

**Pull request merged**